_posts/2024-02-05-cde-customization.adoc

```diff
@@ -0,0 +1,434 @@
+
+TIP: These ConfigMap and Secret should be created in the developer namespace. The variables specified in `data` will be added to the CDEs of owner of the namespace only. Dev Spaces uses namespaces to isolate developers CDEs: it's critical to allow read access to the objects in the namespaces only to the developer owning it.
+
+### Projects files with Kubernetes `ConfigMaps` and `Secrets`
+
+Regular files can be projected into CDEs using `ConfigMaps` and `Secrets` too. In this case the required annotation is `controller.devfile.io/mount-as: file`.
+
+When projecting files, there are two extra annotations read by Eclipse Che: `controller.devfile.io/mount-path: <container-path>` specifies the projected file folder, and `controller.devfile.io/mount-access-mode` specifies the projected file access mode.
+
+The following `ConfigMap` projects a Visual Studio Code `settings.json` file (remote VS Code data folder is `/checode/remote/data/machine`) that specifies a couple of Visual Studio Code Terminal settings:
+
+```bash
+kubectl apply -f - << EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vscode-settings
+  labels:
+    controller.devfile.io/mount-to-devworkspace: "true"
+    controller.devfile.io/watch-configmap: "true"
+  annotations:
+    controller.devfile.io/mount-as: file
+    controller.devfile.io/mount-path: /checode/remote/data/Machine/
+    controller.devfile.io/mount-access-mode: "511"
+data:
+  settings.json: |
+    {
+        "terminal.integrated.copyOnSelection": true,
+        "terminal.integrated.cursorBlinking": true
+    }
+EOF
+```
+
+The following `Secret` projects file `credentials` in fodler `/home/user/.aws`:
+
+```bash
+kubectl apply -f - << EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: aws-credentials
+  labels:
+    controller.devfile.io/mount-to-devworkspace: "true"
+    controller.devfile.io/watch-secret: "true"
+  annotations:
+    controller.devfile.io/mount-as: file
+    controller.devfile.io/mount-path: /home/user/.aws/
+    controller.devfile.io/mount-access-mode: "511"
+stringData:
+  credentials: |
+     [default]
+     aws_access_key_id     = ABCDEFGHILMNOPQRSTUVZ
+     aws_secret_access_key = ABCDEFGHILMNOPQRSTUVZ123456789+abcdefg
+EOF
+```
+
+### Override defaults configurations using a DevWorkspace Operator Configuration
+
+In this last section we want to show a last mechanism to override CDEs default properties. In this case using a https://doc.crds.dev/github.com/devfile/devworkspace-operator[DevWorkspaceOperatorConfig Custom Resource].
+
+`DevWorkspaceOperatorConfig` objects specify advanced CDE properties such as the `Pod` `schedulerName`, whether the `/home/user` folder will persist after a restart or not, and many more.
+
+Applying a particular `DevWorkspaceOperatorConfig` to a CDEs takes 2 steps:
+1. Create the `DevWorkspaceOperatorConfig` custom resource in the Kubernetes cluster
```

Comment:
```suggestion

1. Create the `DevWorkspaceOperatorConfig` custom resource in the Kubernetes cluster
```

Reply:
The numbered list is not being displayed correctly, I think the newline should fix it:

<img width="786" alt="image" src="https://github.com/eclipse-che/blog/assets/83611742/36f4ff1a-0204-4e4c-8e74-b0507018c225">